ciao-controller/command.go

```diff
@@ -129,7 +136,12 @@ func (c *controller) confirmTenantRaw(tenantID string) error {
 			return err
 		}
 
-		if tenant.CNCIIP == "" {
+		i, err := c.ds.GetInstance(tenant.CNCIID)
```

Comment:
it looks like c.launchCNCI is supposed to guarantee that the datastore contains the instance for the CNCI. As that is happening synchronously it would be better for c.launchCNCI to return that instance rather than require the side effect of pulling it from the datastore.